File: internal/model/partner.go
package model

const (
	SupplierStatusActive   = 1
	SupplierStatusInactive = 0
)

type Supplier struct {
	BaseModel
	Name    string `bun:"name,notnull" json:"name"`
	Code    string `bun:"code,unique" json:"code"`
	Contact string `bun:"contact" json:"contact"`
	Phone   string `bun:"phone" json:"phone"`
	Email   string `bun:"email" json:"email"`
	Address string `bun:"address" json:"address"`
	Status  int    `bun:"status,notnull" json:"status"`
}

func (s *Supplier) TableName() string {
	return "suppliers"
}

func (s *Supplier) IsActive() bool {
	return s.Status == SupplierStatusActive
}

const (
	CustomerStatusActive   = 1
	CustomerStatusInactive = 0
)

type Customer struct {
	BaseModel
	Name    string `bun:"name,notnull" json:"name"`
	Code    string `bun:"code,unique" json:"code"`
	Contact string `bun:"contact" json:"contact"`
	Phone   string `bun:"phone" json:"phone"`
	Email   string `bun:"email" json:"email"`
	Address string `bun:"address" json:"address"`
	Status  int    `bun:"status,notnull" json:"status"`
}

func (c *Customer) TableName() string {
	return "customers"
}

func (c *Customer) IsActive() bool {
	return c.Status == CustomerStatusActive
}

File: web/src/types/partner.ts
export interface Supplier {
  id: number
  name: string
  code: string
  contact: string
  phone: string
  email: string
  address: string
  status: number
  created_at: string
  updated_at: string
}

export interface CreateSupplierRequest {
  name: string
  code?: string
  contact?: string
  phone?: string
  email?: string
  address?: string
  status?: number
}

export interface UpdateSupplierRequest {
  name?: string
  code?: string
  contact?: string
  phone?: string
  email?: string
  address?: string
  status?: number
}

export interface Customer {
  id: number
  name: string
  code: string
  contact: string
  phone: string
  email: string
  address: string
  status: number
  created_at: string
  updated_at: string
}

export interface CreateCustomerRequest {
  name: string
  code?: string
  contact?: string
  phone?: string
  email?: string
  address?: string
  status?: number
}

export interface UpdateCustomerRequest {
  name?: string
  code?: string
  contact?: string
  phone?: string
  email?: string
  address?: string
  status?: number
}
